### PR TITLE
Model creation cc

### PR DESCRIPTION
**Pull request by carlosCabrera2**
Created the Model
CC @inaki92 @sarisiza @chr1s4ng3l

### CODE REVIEW

**Comment by inaki92** [comment]
@carlosCabrera2 please add the reviewers to the PR in the left side there is that option.

File: app/build.gradle [hunk 1]
```diff
@@ -1,16 +1,18 @@
 plugins {
     id 'com.android.application'
     id 'org.jetbrains.kotlin.android'
+
```
Comment (inaki92) [inline app/build.gradle:4]: No extra spaces here

File: app/src/main/java/com/example/thecocktailsapp/model/CocktailIInfo/CocktailInfo.kt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package com.example.thecocktailsapp.model.CocktailIInfo
+
+
+import com.squareup.moshi.Json
+import com.squareup.moshi.JsonClass
+
+@JsonClass(generateAdapter = true)
+data class CocktailInfo(
+    @Json(name = "drinks")
+    val drinks: List<DrinkXX>? = listOf()
```
Comment (inaki92) [inline app/src/main/java/com/example/thecocktailsapp/model/CocktailIInfo/CocktailInfo.kt:10]: Does this need to be nullable? and also let's have a better name for the DrinkXX

File: app/src/main/java/com/example/thecocktailsapp/model/CocktailIInfo/DrinkXX.kt [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    @Json(name = "strAlcoholic")
+    val strAlcoholic: String? = null,
+    @Json(name = "strCategory")
+    val strCategory: String? = null,
+    @Json(name = "strCreativeCommonsConfirmed")
+    val strCreativeCommonsConfirmed: String? = null,
+    @Json(name = "strDrink")
+    val strDrink: String? = null,
+    @Json(name = "strDrinkAlternate")
+    val strDrinkAlternate: Any? = null,
```
Comment (inaki92) [inline app/src/main/java/com/example/thecocktailsapp/model/CocktailIInfo/DrinkXX.kt:22]: What is Any? let's define the correct data type

File: app/src/main/java/com/example/thecocktailsapp/model/CocktailIInfo/DrinkXX.kt [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    @Json(name = "strIBA")
+    val strIBA: String? = null,
+    @Json(name = "strImageAttribution")
+    val strImageAttribution: String? = null,
+    @Json(name = "strImageSource")
+    val strImageSource: String? = null,
+    @Json(name = "strIngredient1")
+    val strIngredient1: String? = null,
+    @Json(name = "strIngredient10")
+    val strIngredient10: Any? = null,
```
Comment (inaki92) [inline app/src/main/java/com/example/thecocktailsapp/model/CocktailIInfo/DrinkXX.kt:36]: same here change Any to the real data type

File: app/src/main/java/com/example/thecocktailsapp/model/CocktailIInfo/DrinkXX.kt [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    @Json(name = "strMeasure7")
+    val strMeasure7: Any? = null,
+    @Json(name = "strMeasure8")
+    val strMeasure8: Any? = null,
+    @Json(name = "strMeasure9")
+    val strMeasure9: Any? = null,
+    @Json(name = "strTags")
+    val strTags: String? = null,
+    @Json(name = "strVideo")
+    val strVideo: Any? = null
```
Comment (inaki92) [inline app/src/main/java/com/example/thecocktailsapp/model/CocktailIInfo/DrinkXX.kt:110]: If there is a video on youtube...  let's think about playing the video in EXO PLAYER

File: app/src/main/java/com/example/thecocktailsapp/model/CocktailsByName/CocktailsByName.kt [hunk 6]
```diff
@@ -0,0 +1,11 @@
+package com.example.thecocktailsapp.model.CocktailsByName
+
+
+import com.squareup.moshi.Json
+import com.squareup.moshi.JsonClass
+
+@JsonClass(generateAdapter = true)
+data class CocktailsByName(
+    @Json(name = "drinks")
+    val drinks: List<Drink?>? = null
```
Comment (inaki92) [inline app/src/main/java/com/example/thecocktailsapp/model/CocktailsByName/CocktailsByName.kt:10]: The Drink as nullable can be omitted

File: app/src/main/java/com/example/thecocktailsapp/model/CocktailsByName/Drink.kt [hunk 7]
```diff
@@ -0,0 +1,111 @@
+package com.example.thecocktailsapp.model.CocktailsByName
+
+
+import com.squareup.moshi.Json
+import com.squareup.moshi.JsonClass
+
+@JsonClass(generateAdapter = true)
+data class Drink(
```
Comment (inaki92) [inline app/src/main/java/com/example/thecocktailsapp/model/CocktailsByName/Drink.kt:8]: THis looks the same as the DrinkXX, find out if it is needed, if not remove it

**Review by inaki92 — APPROVED** [review]
Make the changes and push the new code.

Also, make sure to use the correct naming for classes.

File: app/build.gradle [hunk 8]
```diff
@@ -34,11 +36,21 @@ android {
...
     implementation 'androidx.appcompat:appcompat:1.6.0'
     implementation 'com.google.android.material:material:1.8.0'
     implementation 'androidx.constraintlayout:constraintlayout:2.1.4'
+    implementation 'com.squareup.moshi:moshi-kotlin:1.14.0'
+
+    implementation 'com.google.code.gson:gson:2.10.1'
+    implementation 'com.squareup.retrofit2:converter-gson:2.9.0'
+    implementation 'com.squareup.retrofit2:retrofit:2.9.0'
+
+    implementation 'com.google.dagger:dagger-android:2.44.2'
```
Comment (sarisiza) [inline app/build.gradle:49]: This should be hilt instead of dagger

**Review by sarisiza — APPROVED** [review]
Check comments and check
We should only have a single Drink and Cocktails data class

File: app/src/main/java/com/example/thecocktailsapp/model/CocktailIInfo/DrinkXX.kt [hunk 9]
```diff
@@ -0,0 +1,111 @@
+package com.example.thecocktailsapp.model.CocktailIInfo
+
+
+import com.squareup.moshi.Json
+import com.squareup.moshi.JsonClass
+
+@JsonClass(generateAdapter = true)
+data class DrinkXX(
```
Comment (chr1s4ng3l) [inline app/src/main/java/com/example/thecocktailsapp/model/CocktailIInfo/DrinkXX.kt:8]: Duplicated model.
(DrrinkXX and Drink)

File: app/src/main/java/com/example/thecocktailsapp/model/CocktailsByName/Drink.kt (hunk 7 above)
Comment (chr1s4ng3l) [inline app/src/main/java/com/example/thecocktailsapp/model/CocktailsByName/Drink.kt:8]: Duplicated model.
(DrrinkXX and Drink)

**Review by chr1s4ng3l — CHANGES_REQUESTED** [review]
We need to set only the necessary data we do not need extra data unusable.

**Review by chr1s4ng3l — APPROVED** [review]

**Review by carlosCabrera2 — COMMENTED** [review]
i made the changes

**Pull request merged**